packages/@blockprotocol/hook/src/react.ts

```diff
@@ -243,50 +240,45 @@ export const useHook = <T extends HTMLElement>(
 
       existingHookRef.current = hook;
 
-      teardownPromise
-        .then(() => {
-          if (service.destroyed || controller.signal.aborted) {
-            return;
-          }
-
-          return service
-            .hook({
-              data: {
-                hookId: hook.id,
-                entityId,
-                node,
-                type,
-                path,
-              },
-            })
-            .then((response) => {
-              if (!controller.signal.aborted) {
-                if (response.errors) {
-                  if (
-                    response.errors.length === 1 &&
-                    response.errors[0]?.code === "NOT_IMPLEMENTED"
-                  ) {
-                    const teardown = fallbackRef.current(node);
+      void service
+        .hook({
+          data: {
+            hookId: hook.id,
+            entityId,
+            node,
+            type,
+            path,
+          },
+        })
+        .then((response) => {
+          if (!controller.signal.aborted) {
+            if (response.errors) {
+              const firstError = response.errors[0];
+              if (firstError?.code === "NOT_IMPLEMENTED") {
+                const teardown = fallbackRef.current(node);
 
-                    hook.teardown = async () => {
-                      controller.abort();
-                      teardown?.();
-                    };
-                  } else {
-                    // eslint-disable-next-line no-console
-                    console.error(response.errors);
-                    throw new Error("Unknown error in hook");
-                  }
-                } else if (response.data) {
-                  hook.id = response.data.hookId;
+                hook.teardown = async () => {
+                  controller.abort();
+                  teardown?.();
+                };
+              } else if (firstError?.code === "NOT_FOUND") {
+                const errMsg = `Hook with id ${hook.id} was not found by embedding application`;
+                if (node === null) {
+                  // don't throw if the request was for hook deletion – the embedding app can't find the hook, things can continue
+                  // eslint-disable-next-line no-console -- useful for debugging
+                  console.warn(`${errMsg} – no hook to remove`);
+                } else {
+                  throw new Error(errMsg);
                 }
+              } else {
+                // eslint-disable-next-line no-console
+                console.error(response.errors);
+                throw new Error("Unknown error in hook");
               }
-            });
-        })
-        .catch((err) => {
-          catchError(() => {
-            throw err;
-          });
```

Comment:
Blocking: this needs to be here so React catches the errors thrown above – unsure why it's been removed.

Reply:
Thank you, restored in https://github.com/blockprotocol/blockprotocol/pull/707/commits/8cbb862915ec99adce6ca230d209ec5a07b5a8af